fix(semantic): remove inherting `ScopeFlags::Modifiers` from parent scope

Dunqing opened the pull request:
close: #7900

After #4283 changed, we don't need to inherit `ScopeFlags` from the `constructor`, `set`, `get` anymore, I think this is a logic of forgetting to remove

Dunqing commented:
* **#7932** <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/7932?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a> 👈 <a href="https://app.graphite.dev/github/pr/oxc-project/oxc/7932?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* `main`
<hr />

## How to use the Graphite Merge Queue

Add either label to this PR to merge it via the merge queue:
- <kbd>0-merge</kbd> - adds this PR to the back of the merge queue
- <kbd>hotfix</kbd> - for urgent hot fixes, skip the queue and merge this PR next

You must have a Graphite account in order to use the merge queue. Sign up using [this link](https://app.graphite.dev/invite/github/oxc-project?ref=merge-queue-instructions-in-stack-comment&prId=7932).

<h2></h2>
<sub>

An organization admin has enabled the [Graphite Merge Queue](https://graphite.dev/docs/graphite-merge-queue) in this repository.

</sub>

<sub>

Please do not merge from GitHub as this will restart CI on PRs being processed by the merge queue.

</sub>

<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.
<!-- Current dependencies on/for this PR: -->

Dunqing commented:
https://github.com/oxc-project/oxc/actions/runs/12347529516/job/34454785628?pr=7932
I just realized why we still need to inherit when I saw the failed linter test, But I think we should fix this test in that linter rule rather than by inheriting `ScopeFlags`.

Boshen commented on crates/oxc_semantic/examples/semantic.rs:
```diff
@@ -39,6 +39,8 @@ fn main() -> std::io::Result<()> {
         .with_check_syntax_error(true)
         .build(&program);
 
+    println!("{:?}", semantic.semantic.scopes());
```
```suggestion
```

Boshen commented on crates/oxc_semantic/examples/semantic.rs:
```diff
@@ -39,6 +39,8 @@ fn main() -> std::io::Result<()> {
         .build(&program);
 
+    println!("{:?}", semantic.semantic.scopes());
+
```
```suggestion
```

overlookmotel commented:
### Merge activity

* **Dec 16, 6:25 AM EST**: The merge label '0-merge' was detected. This PR will be added to the [Graphite merge queue](https://app.graphite.dev/merges?org=oxc-project&repo=oxc) once it meets the requirements.

Dunqing committed:
fix(semantic): remove inherting `ScopeFlags::Modifier` from parent scope (#7932)

close: #7900

After #4283 changed, we don't need to inherit `ScopeFlags` from the `constructor`, `set`, `get` anymore, I think this is a logic of forgetting to remove
